src/thor/trace_route_action.cc

```diff
@@ -562,9 +596,10 @@ thor_worker_t::map_match(Api& request) {
 
             // we get the time up to the last edge before this end edge if any. we also remove
             // the turn cost at the begging of this edge if there is any
-            double trim_end = trivial_leg && begin_trimmed
-                                  ? 0
+            double trim_end = leg_end == path_edges.cbegin()
```

Comment:
detect if the leg end is the first leg in the path, it not, we get the time up to when this edge end

Reply:
i havent had a chance to grok this change here. i know i know, i wrote the code, but i need to get my head back into this computation to understand what the change is here. the thing this was trying to worry about was when the leg was on a single edge. im just wondering how that would still work if the decision about what to trim from the end is based on whether its the first edge in the path or not. specifically because single edge legs can be anywhere in the path, not just the beginning of it. sorry taking me a while to understand this one